lib/middleware.py

```diff
@@ -11,15 +10,11 @@ class LocaleMiddleware(MiddlewareMixin):
     response_redirect_class = HttpResponseRedirect
 
     def process_request(self, request):
-        urlconf = getattr(request, 'urlconf', settings.ROOT_URLCONF)
-        i18n_patterns_used, prefixed_default_language = is_language_prefix_patterns_used(urlconf)
-        language = translation.get_language_from_request(request, check_path=i18n_patterns_used)
-        language_from_path = translation.get_language_from_path(request.path_info)
-        if not language_from_path:
-            if request.user.is_authenticated:
-                language = request.user.userprofile.lang
-            elif not prefixed_default_language and i18n_patterns_used: 
-                language = settings.LANGUAGE_CODE
+        language = translation.get_language_from_request(request)
+        if request.user.is_authenticated:
```

Comment:
Maybe also check that `request.user` exists, might be. I guess it would make sense to check how django code middlewares handle that internally. (i.e. do they raise an exception if there isn't one or do they have a check `hasattr(reqyest, 'user')`)

Reply:
I think that `request.user` always exists when there is an authentication system, it's just `AnonymousUser` when not logged in.

https://docs.djangoproject.com/en/3.0/ref/contrib/auth/#django.contrib.auth.models.User.is_authenticated

Reply:
I was thinking django would do something like this https://github.com/django/django/blob/master/django/contrib/auth/middleware.py#L17

Reply:
So I should add that also to my `process_request`?

Reply:
I guess it would be a bit nicer way to crash, than an AttributeError